### ISSUE TITLE
Chyt can't read decimal

### ISSUE THREAD

**Issue by ykuc** [ytsaurus/ytsaurus#863]
If I try read yt type Decimal (example Decimal(35, 15)) use CHYT
<img width="248" alt="image" src="https://github.com/user-attachments/assets/47ab2c55-39f7-4f2a-9432-3f40a5b54399">
I have an error
`ClickHouse type "Nullable(Decimal(38, 15))" is not representable as YT type: maximum decimal precision in YT is 35`

**Comment by DimasKovas** [comment]
> If I try read yt type Decimal (example Decimal(35, 15)) use CHYT <img alt="image" width="248" src="https://private-user-images.githubusercontent.com/140019825/370662346-47ab2c55-39f7-4f2a-9432-3f40a5b54399.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyNTg2MjMsIm5iZiI6MTcyNzI1ODMyMywicGF0aCI6Ii8xNDAwMTk4MjUvMzcwNjYyMzQ2LTQ3YWIyYzU1LTM5ZjctNGYyYS05NDMyLTNmNDBhNWI1NDM5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNVQwOTU4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MDAxZGNkYmRhYzRmNGNkZGRiYzdlMjAyNjE4NTgyYmNjMTJjNThlODhkZmY1OGU2MDJiZGIwZTZhZGE1MTZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oF88aKTunPonXrkcPxjI9dpFcn94skE0LhUJZ6ljmW8"> I have an error `ClickHouse type "Nullable(Decimal(38, 15))" is not representable as YT type: maximum decimal precision in YT is 35`

Please provide a query to reproduce the problem.

I see one flaw in a decimal conversion that can lead to such errors I'll fix it, but it's not related to reading from table, and `select * from table_with_decimals` works fine for me.

**Comment by zlobober** [comment]
This is probably related to the fact that YT decimal is indeed shorter by 3 decimal digits currently.
There is actually a PR fixing this issue:  https://github.com/ytsaurus/ytsaurus/pull/798

**Comment by zlobober** [comment]
Still, it will take quite a long time to wait for the PR above to merge and become available in the stable branches, so I'd suggest casting to String on ClickHouse side and saving such values as strings (or casting to Decimal(35, 15) if the values actually fit into that scale).

**Comment by ykuc** [comment]
> Пожалуйста, отправьте запрос, чтобы воспроизвести проблему.
Проверил еще раз. Проблема возникает при группировке данных. Если просто select - все хорошо, если группировка то выходит ошибка
```
INSERT INTO `sandbox/test_dec`
select cast('2.1' as Decimal(35,15)) as dec
```
![image](https://github.com/user-attachments/assets/ea9d282b-ee55-4f80-9cc7-67726fdc5e24)
```
select sum(usage_quantity)
from `sandbox/test_dec`
```
![image](https://github.com/user-attachments/assets/0dd71e91-3efb-4c45-bac0-0af9baf9ae5b)

**Comment by ykuc** [comment]
Chyt пытается увеличить размерность при группировки?
вроде такое раньше было в старом ch

А почему такой проблемы не было в ya?

**Comment by DimasKovas** [comment]
> Is Chyt trying to increase the dimension when grouping?

It's not chyt, it's ClickHouse's engine. We don't have any chyt-specific logic here.

> Why wasn't there such a problem in ya?

The problem is related to the query tracker. If you use the HTTP protocol to make queries, there will be no error. Yandex has not yet shipped the query tracking component to production (but we have plans to do so in the near future). We use an alternative internal solution that uses the HTTP protocol and does not suffer from this problem.

**Comment by zlobober** [comment]
Will there be a fix for QT <-> CHYT interaction?

**Comment by DimasKovas** [comment]
> Will there be a fix for QT <-> CHYT interaction?

I've already fixed it yesterday in master and 2.16 (the OS announcement with release notes will be ready soon).

That's why the issue has been closed.
